napari/layers/labels/_labels_mouse_bindings.py

```diff
@@ -16,7 +37,15 @@ def paint(layer, event):
             last_cursor_coord, layer.coordinates, layer.brush_size
         )
         for c in interp_coord:
-            layer.paint(c, layer.selected_label, refresh=False)
+            if layer._mode == Mode.PAINT:
+                layer.paint(c, layer.selected_label, refresh=False)
+            elif (
+                layer._mode == Mode.FILL
+                and layer._value != layer.selected_label
+            ):
+                layer.fill(
+                    layer.coordinates, layer._value, layer.selected_label
```

Comment:
Inside the for loop this should be
```suggestion
                    c, layer._value, layer.selected_label
```
to use the "interpolated coordinates" instead of just the current clicked coordinate

Reply:
addressed the interpolation and clip on coordinates, nice catch!